collated/main/rushan-khor.md

```diff
@@ -0,0 +1,319 @@
+    public static final String COMMAND_ALIAS = "b"; // shorthand equivalent alias
+
+    public static final String MESSAGE_USAGE = COMMAND_WORD
+            + ": Deletes all persons with the given tags.\n"
+            + COMMAND_ALIAS + ": Shorthand equivalent for batch delete. \n"
+            + "Parameters: TAG (must be a tag that at least one person has)\n"
+            + "Example 1: " + COMMAND_ALIAS + " friends \n"
+            + "Example 2: " + COMMAND_WORD + " colleagues";
+
+    public static final String MESSAGE_BATCH_DELETE_SUCCESS = "Deleted Persons with Tags: %1$s";
+
+    private final Set<Tag> tagsToDelete;
+
+    public BatchCommand(Set<Tag> tagsToDelete) {
+        this.tagsToDelete = tagsToDelete;
+    }
+
+
+    @Override
+    public CommandResult executeUndoableCommand() throws CommandException {
+        try {
+            model.deletePersonsByTags(tagsToDelete);
+        } catch (PersonNotFoundException e) {
+            throw new CommandException("The target person cannot be missing");
+        }
+
+        return new CommandResult(String.format(MESSAGE_BATCH_DELETE_SUCCESS, tagsToDelete));
+    }
+
+    @Override
+    public boolean equals(Object other) {
+        return other == this // short circuit if same object
+                || (other instanceof BatchCommand // instanceof handles nulls
+                && this.tagsToDelete.equals(((BatchCommand) other).tagsToDelete)); // state check
+    }
+}
+```
+###### \java\seedu\address\logic\commands\CopyCommand.java
+``` java
+    @Override
+    public CommandResult execute() throws CommandException {
+        String targetEmail = getTargetEmail();
+        String commandResultMessage = "";
+
+        boolean emailIsValid = isEmailValid(targetEmail);
+        if (emailIsValid) {
+            putIntoClipboard(targetEmail);
+            commandResultMessage = String.format(MESSAGE_COPY_PERSON_SUCCESS, targetPerson.getName());
+        } else {
+            commandResultMessage = String.format(MESSAGE_COPY_PERSON_EMPTY, targetPerson.getName());
+        }
+
+        return new CommandResult(commandResultMessage);
+    }
+
+    /**
+     * Gets the target person's email address.
+     * @return     the email address of the person at the list {@code targetIndex}
+     * @exception  CommandException if the {@code targetIndex}
+     *             argument is greater than or equal to the {@code lastShownList} size.
+     */
+    public String getTargetEmail() throws CommandException {
+        List<ReadOnlyPerson> lastShownList = model.getFilteredPersonList();
+        boolean indexIsOutOfBounds = targetIndex.getZeroBased() >= lastShownList.size();
+        if (indexIsOutOfBounds) {
+            throw new CommandException(Messages.MESSAGE_INVALID_PERSON_DISPLAYED_INDEX);
+        }
+
+        targetPerson = lastShownList.get(targetIndex.getZeroBased());
+        return targetPerson.getEmail().toString();
+    }
+
+    public boolean isEmailValid(String email) {
```

Comment:
Missing header comment. All non trivial methods should have javadoc format header comments.
This method seems to be handling two cases?